File: src/main/java/com/rotpaddon/exampleaddon/power/impl/nonstand/type/tutorial/TutorialData.java
package com.rotpaddon.exampleaddon.power.impl.nonstand.type.tutorial;

import com.github.standobyte.jojo.power.impl.nonstand.TypeSpecificData;
import com.rotpaddon.exampleaddon.network.AddonPackets;
import com.rotpaddon.exampleaddon.network.server.TutorialDataPacket;

import net.minecraft.entity.LivingEntity;
import net.minecraft.entity.player.ServerPlayerEntity;
import net.minecraft.nbt.CompoundNBT;

public class TutorialData extends TypeSpecificData {
    public boolean isArmored;
    
    public void setIsArmored(boolean isArmored) {
        LivingEntity user = power.getUser();
        this.isArmored = isArmored;
        if (!user.level.isClientSide()) {
        	 AddonPackets.sendToClientsTrackingAndSelf(new TutorialDataPacket(user.getId(), isArmored), user);
        }
    }

    @Override
    public CompoundNBT writeNBT() {
        CompoundNBT nbt = new CompoundNBT(); // Better to save data, like states or some other values.
        nbt.putBoolean("isArmored", isArmored);
        return nbt;
    }

    @Override
    public void readNBT(CompoundNBT nbt) {
    	this.isArmored = nbt.getBoolean("isArmored");
    }

    @Override
    public void syncWithUserOnly(ServerPlayerEntity user) { // Syncs with user (with packets)
		AddonPackets.sendToClientsTrackingAndSelf(new TutorialDataPacket(user.getId(), isArmored), user);
    }

    @Override
    public void syncWithTrackingOrUser(LivingEntity user, ServerPlayerEntity entity) {
   	 	AddonPackets.sendToClientsTrackingAndSelf(new TutorialDataPacket(user.getId(), isArmored), user);
    }
}

File: src/main/kotlin/com/rotpaddon/exampleaddon/power/impl/nonstand/type/tutorial/TutorialData.kt
package com.rotpaddon.exampleaddon.power.impl.nonstand.type.tutorial

import com.github.standobyte.jojo.power.impl.nonstand.TypeSpecificData
import com.rotpaddon.exampleaddon.network.AddonPackets
import com.rotpaddon.exampleaddon.network.server.TutorialDataPacket
import net.minecraft.entity.LivingEntity
import net.minecraft.entity.player.ServerPlayerEntity
import net.minecraft.nbt.CompoundNBT

class TutorialData : TypeSpecificData() {
    var isArmored: Boolean = false

    fun setIsArmored(isArmored: Boolean) {
        val user = power.user
        this.isArmored = isArmored
        if (!user.level.isClientSide) {
            AddonPackets.sendToClientsTrackingAndSelf(TutorialDataPacket(user.id, isArmored), user)
        }
    }

    override fun writeNBT(): CompoundNBT {
        val nbt = CompoundNBT()
        nbt.putBoolean("isArmored", isArmored)
        return nbt
    }

    override fun readNBT(nbt: CompoundNBT) {
        this.isArmored = nbt.getBoolean("isArmored")
    }

    override fun syncWithUserOnly(user: ServerPlayerEntity) {
        AddonPackets.sendToClientsTrackingAndSelf(TutorialDataPacket(user.id, isArmored), user)
    }

    override fun syncWithTrackingOrUser(user: LivingEntity, entity: ServerPlayerEntity) {
        AddonPackets.sendToClientsTrackingAndSelf(TutorialDataPacket(user.id, isArmored), user)
    }
}